unity/Assets/Scripts/ColorChanger.cs

```diff
@@ -1,84 +1,214 @@
 ﻿using System.Linq;
 using UnityEngine;
+using System.Collections;
+using System.Collections.Generic;
+using Random = UnityEngine.Random;
  
 public class ColorChanger : MonoBehaviour {
-    // Define all material types
-    public Material[] alarmClockMaterials;
-    public Material[] appleMaterials;
-    public Material[] basketballMaterials;
-    public Material[] bowlMaterials;
-    public Material[] garbageBinMaterials;
-    public Material[] houseplantMaterials;
-    public Material[] pillowMaterials;
-    public Material[] sprayBottleMaterials;
-
-    public Material[] boxMaterials;
-    public Material[] cellphoneMaterials;
-    public Material[] cupMaterials;
-    public Material[] floorLampMaterials;
-    public Material[] penPencilMaterials;
-    public Material[] plateMaterials;
-    public Material[] potMaterials;
-    public Material[] statueMaterials;
-    public Material[] watchMaterials;
-
-    public Material[] armchairMaterials;
-    public Material[] bedMaterials;
-    public Material[] chairMaterials;
-    public Material[] coffeeTableMaterials;
-    public Material[] deskMaterials;
-    public Material[] diningTableMaterials;
-    public Material[] dresserMaterials;
-    public Material[] officeChairMaterials;
-    public Material[] shelvingUnitMaterials;
-    public Material[] sideTableMaterials;
-    public Material[] sofaMaterials;
-
-    public Material[] ceramicMaterials;
-    public Material[] fabricMaterials;
-    public Material[] glassMaterials;
-    public Material[] lightMaterials;
-    public Material[] metalMaterials;
-    public Material[] miscMaterials;
-    public Material[] plasticMaterials;
-    public Material[] woodMaterials;
-
-    Material[] targetMaterials;
-    Material[] backgroundMaterials;
-    Material[] furnitureMaterials;
-    Material[] quickMaterials;
-
-    Material[] allMaterials;
-    Color[] origColors;
+    public Material[] alarmClockMaterials,
```

Comment:
Materials were dragged here and stored in the prefab. For a future iteration we could create a json dictionary from object class to list of materials, and load them dynamically with asset `Resources.Load` or `AssetDatabase.LoadAssetAtPath`. Changing this text file would be easier and visible in PRs.

Reply:
This is true. Although the materials are stored in pretty random file locations, where metal materials alone might be spread out between, say. 20 different folders and dozens of different paths.

So there would have to be a major big refactor/cleanup before that would work and I'm not sure if there's an immediate benefit to this approach speed wise.